test/parallel/test-runner-bail.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+require('../common');
+const fixtures = require('../common/fixtures');
+const tmpdir = require('../common/tmpdir');
+const { describe, it } = require('node:test');
+const { spawnSync } = require('node:child_process');
+const assert = require('node:assert');
+
+const testFile = fixtures.path('test-runner/bail/bail.js');
+const errorTestFile = fixtures.path('test-runner/bail/error.js');
+
+tmpdir.refresh();
+
+describe('maintain errors', () => {
+  it('should exit at assertion failure', () => {
+    const child = spawnSync(process.execPath, ['--test', '--test-bail', errorTestFile]);
+    assert.strictEqual(child.stderr.toString(), '');
+    assert.match(child.stdout.toString(), /failureType: 'testCodeFailure'/);
```

Comment:
better to use a snapshot test here

Reply:
you mean with `tap` `matchSnapshot`?

Reply:
no - see https://github.com/nodejs/node/blob/main/test/parallel/test-runner-output.mjs

Reply:
It looks like this request is still pending. I think addressing this would make this PR easier to review since the test runner output will shown in the snapshot.

Reply:
should I snapshot also spec?

Reply:
one reporter is enough

Reply:
done 😃